### ISSUE TITLE
generator-star-spacing warning on async function

### ISSUE THREAD

**Issue by rightaway** [eslint/eslint#4415]
I'm using eslint 1.9.0 with babel-eslint parser. I have generator-star-spacing option set to `before` in .eslintrc.

A function signature like `export async function test(b, c) {` gives an error `Unexpected space after *. (generator-star-spacing)`, and shows the error happening in column 23. My IDE underlines the 24th character, which in this case would be the `e` in the function name `test`. For all similar functions it's the same thing, same error and in the second character of the function name.

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
I think your example got cut off, since I don't see a generator function anywhere. However in general, ESLint doesn't support experimental features directly. As such we can't guarantee that our rules will work with experimental syntax. You might want to file an issue with [eslint-plugin-babel](https://github.com/babel/eslint-plugin-babel) if they do not already support this.
If you can reproduce this case without using async, please reopen this issue.

**Comment by rightaway** [comment]
There isn't a generator or `*` anywhere in that function, so it's quite strange.

**Comment by ilyavolodin** [comment]
Ahhh.. That looks like a bug in `babel-eslint` for some reason function is marked as `generator`. @hzoo I assume it shouldn't be, right?

**Comment by hzoo** [comment]
We had a bunch of things we added so eslint wouldn't break on other rules that I slowly removed as more es6 and other stuff got added.

https://github.com/babel/acorn-to-esprima/blob/master/src/toAST.js#L69-L78

It's basically

```js
// async function as generator
    if (this.isFunction()) {
      if (node.async) node.generator = true;
    }
```

which we can remove but it might affect how other rules work not sure so I left it in at the time.

**Comment by hzoo** [comment]
Published acorn-to-esprima@1.0.6 which removes the transform above so you just need to update.

**Comment by hzoo** [comment]
Getting a lot of complaints about other rules breaking so I guess reverting that fix in 1.0.7. Not sure what to do.
